src/main/CommandLine.cpp

```diff
@@ -1926,6 +1928,11 @@ runApplyLoad(CommandLineArgs const& args)
 
                 for (size_t i = 0; i < 100; ++i)
                 {
+                    auto checkMergingWork =
+                        app.getWorkScheduler().executeWork<CheckMergingWork>();
```

Comment:
This doesn't need to be a Work class, right? `resolveAnyReadyFutures` is a synchronous function, so we're not taking any advantage of the Work class which is designed to execute small-grained async tasks until completion. Perhaps we can simply call `app.getBucketManager().getBucketList().resolveAnyReadyFutures();` here?

Reply:
Ah, we want to resolve _all_ futures here, so `resolveAnyReadyFutures` is not sufficient, so maybe we just need a synchronous function `resolveAllFutures` in the BucketList class (we currently have a function like that in HistoryArchive class, so maybe we need to refactor it a bit to avoid duplication)

Reply:
actually, you can probably just call `app.getLedgerManager().getLastClosedLedgerHAS().resolveAllFutures()`